lib/connect/index.js

```diff
@@ -147,8 +147,9 @@ function connect (brokerUrl, opts) {
 
     return protocols[opts.protocol](client, opts)
   }
-
-  return new MqttClient(wrapper, opts)
+  var client = new MqttClient(wrapper, opts)
+  client.on('error', function () { /* Automatically set up client error handling */ })
```

Comment:
>client.on('error', function () { /* Automatically set up client error handling */ }) [](start = 2, length = 84)

Does this prevent the application from faulting if an error is emitted from mqttjs and nobody 'handles' it?

Reply:
yes.

---
In reply to: [407712729](https://github.com/mqttjs/MQTT.js/pull/1076#discussion_r407712729) [](ancestors = 407712729)